File: examples/telegram-ts/hosted.ts
/* Telegram hosted by Caspian, TypeScript. Same handlers as bot.ts; the
   gateway owns inbound. This process never sees a Telegram Update: cx.run()
   polls GET /v1/events and drives the same pipeline.

   Hosted does not mint a BotFather bot. You still pass TELEGRAM_BOT_TOKEN. */
import { Caspian } from "caspian-sdk"
import { register } from "./app.ts"

const token = (process.env.TELEGRAM_BOT_TOKEN ?? "").trim()
const apiKey = (process.env.CASPIAN_API_KEY ?? "").trim()
if (token === "") {
  console.error("Set TELEGRAM_BOT_TOKEN (BotFather → /newbot), then rerun.")
  process.exit(1)
}
if (apiKey === "") {
  console.error("Set CASPIAN_API_KEY, then rerun.")
  process.exit(1)
}

const cx = new Caspian()
await cx.channels.add("telegram", { via: "hosted", bot_token: token })
register(cx)

console.log("hosted telegram (ts) — polling gateway /v1/events")
const results = await cx.run({ apiKey })
for (const result of results) {
  if (!(result as { ok: boolean }).ok) {
    console.error(JSON.stringify((result as { error: unknown }).error))
  }
}

File: examples/telegram/hosted.py
"""Telegram hosted by Caspian. Same handlers as bot.py; the gateway owns inbound.

Hosted does not mint a BotFather bot. You still pass TELEGRAM_BOT_TOKEN.
This process never sees a Telegram Update: cx.run() polls GET /v1/events and
feeds each payload to handle("gateway", …).
"""

from __future__ import annotations

import os

from app import register
from caspian import Caspian

token = os.environ.get("TELEGRAM_BOT_TOKEN", "").strip()
api_key = os.environ.get("CASPIAN_API_KEY", "").strip()
if not token:
    raise SystemExit("Set TELEGRAM_BOT_TOKEN (BotFather → /newbot), then rerun.")
if not api_key:
    raise SystemExit("Set CASPIAN_API_KEY, then rerun.")

cx = Caspian(api_key=api_key)
cx.channels.add("telegram", bot_token=token)
register(cx)

if __name__ == "__main__":
    print("hosted telegram — polling gateway /v1/events", flush=True)
    cx.run()
